icechunk-python/python/icechunk/__init__.py

```diff
@@ -33,6 +34,7 @@
 
 class IcechunkStore(Store, SyncMixin):
     _store: PyIcechunkStore
+    _pickle_preserves_read_only: bool
```

Comment:
I put this on this class, not PyIcechunkStore, since this is the one that handles pickling.